### DISCUSSION TITLE
What is the reason for adding WebSurferAgent as an agent as opposed to a function?

### DISCUSSION THREAD

**Discussion by ColtonBehannon** [microsoft/autogen#1566] [General]
I've been experimenting with WebSurferAgent lately and am wondering what was the reasoning behind including it as its own agent as opposed to a retrievable function to be called by any agent? The concept of including a new agent just to be able to surf the web seems strange to me. Any context on this decision or why it is configured this way?

**Comment by sonichi** [comment]
It's an example of demonstrating the possibility and using it for research purpose. You're welcome to share your ideas or use cases.

**Reply by ColtonBehannon** [reply]
It's very useful. I was more so curious if there is a reason that it's configured as a standalone agent similar to how some groupchat's tend to have a standalone agent that is solely a code interpreter instead of stacking both code writing and executing in one agent. Are there any benefits in the execution that have been noticed by isolating the responsibilities of these "tool" agents? Or is it just for a more natural looking conversation?

**Reply by ekzhu** [reply]
One reason for separating is that it allows natural human intervention before any tool or code execution hence the name "user proxy".
